lib/net/ssh/transport/session.rb

```diff
@@ -126,6 +126,7 @@ def closed?
         # Cleans up (see PacketStream#cleanup) and closes the underlying socket.
         def close
           socket.cleanup
+          Process.kill('TERM', socket.pid) if options[:proxy].is_a?(Net::SSH::Proxy::Command)
```

Comment:
I don't really like this solution. options[:proxy], if anything this should be moved to Proxy::Command. the returned io close method could easily do this kill. But I'd really like to see the concrete use case first